pkg/services/grafana-apiserver/endpoints/request/request.go

```diff
@@ -2,25 +2,66 @@ package request
 
 import (
 	"context"
+	"fmt"
 	"strconv"
+	"strings"
 
 	"k8s.io/apiserver/pkg/endpoints/request"
 )
 
-func OrgIDFrom(ctx context.Context) (int64, bool) {
-	ns := request.NamespaceValue(ctx)
-	return ParseOrgID(ns)
+type NamespaceInfo struct {
+	// OrgID defined in namespace (1 when using stack ids)
+	OrgID int64
+
+	// The cloud stack ID (must match the value is cfg.Settings)
+	StackID string
+
+	// For namespaces that are not org-{id} or stack-{id}
+	Other string
```

Comment:
`Other` as a fallback feels strange to me. what do you think about something like `Raw`, `Original`, or `Value` and always storing the unparsed `request.NamespaceValue(ctx)` string? that info might be useful to have for logging in case parsing breaks in some unexpected way

Reply:
+1 I like Value